src/components/KCodeBlock/KCodeBlock.vue

```diff
@@ -780,491 +710,217 @@ async function copyCode(event: Event): Promise<void> {
+              transition: border $kongponentsTransitionDurTimingFunc;
+              width: 100%;
+            }
+
+            &.line-is-highlighted-match {
+              &::before {
+                border-left: var(--kui-border-width-30, $kui-border-width-30) solid var(--kui-color-border-primary, $kui-color-border-primary);
+              }
+            }
+          }
+        }
+      }
 
-.k-code-block-search-error,
-.k-code-block-search-results {
-  margin-bottom: var(--kui-space-0, $kui-space-0);
-  margin-top: var(--kui-space-0, $kui-space-0);
-}
+      code {
+        @include kCodeBlockTypography;
 
-.k-code-block-search-error {
-  background-color: var(--kui-color-background, $kui-color-background);
-  border: var(--kui-border-width-10, $kui-border-width-10) solid currentColor;
-  border-bottom-left-radius: var(--kui-border-radius-10, $kui-border-radius-10);
-  border-bottom-right-radius: var(--kui-border-radius-10, $kui-border-radius-10);
-  color: var(--kui-color-text-danger, $kui-color-text-danger);
-  font-size: var(--kui-font-size-20, $kui-font-size-20);
-  left: -1px;
-  padding: var(--kui-space-0, $kui-space-0) var(--kui-space-20, $kui-space-20);
-  position: absolute;
-  right: -1px;
-  top: 100%;
-  z-index: 1;
-}
+        color: var(--kui-color-text-neutral-strongest, $kui-color-text-neutral-strongest);
+        display: block;
+        min-width: 0;
+        overflow-x: auto;
+        z-index: 1;
+      }
 
-.k-search-icon {
-  color: var(--kui-color-text-neutral-weak, $kui-color-text-neutral-weak);
-  padding: var(--kui-space-0, $kui-space-0) var(--kui-space-20, $kui-space-20);
+      &.single-line {
+        grid-template-columns: auto;
+        padding-right: var(--kui-space-100, $kui-space-100);
+      }
+    }
 
-  .theme-light {
-    color: var(--kui-color-text-neutral, $kui-color-text-neutral);
+    .code-block-secondary-actions {
+      display: flex;
+      gap: var(--kui-space-40, $kui-space-40);
+      margin-right: var(--kui-space-40, $kui-space-40);
+      margin-top: var(--kui-space-40, $kui-space-40);
+      position: absolute;
+      right: 0;
+      top: 0;
+      z-index: 1;
+    }
   }
 
-  .theme-dark {
-    color: var(--kui-color-text-neutral-weak, $kui-color-text-neutral-weak);
-  }
-}
+  &.theme-dark {
+    background-color: var(--kui-color-background-inverse, $kui-color-background-inverse);
+
+    .code-block-actions {
+      border-bottom-color: var(--kui-color-border-inverse, $kui-color-border-inverse);
+
+      .code-block-search-input {
+        :deep(input) {
+          color: var(--kui-color-text-inverse, $kui-color-text-inverse);
+
+          &::placeholder {
+            color: var(--kui-color-text-neutral-weaker, $kui-color-text-neutral-weaker);
+          }
+        }
+
+        .clear-query-button,
+        .code-block-search-icon {
+          color: var(--kui-color-text-neutral-weaker, $kui-color-text-neutral-weaker) !important;
+        }
+
+        .clear-query-button[role="button"] {
```

Comment:
(Needed for the other suggestion I made related to this element)

```suggestion
        .clear-query-button {
```